ts_src/note.ts

```diff
@@ -0,0 +1,95 @@
+            title: this.name,
+            autoOpen: false,
+            width: 'auto',
+            buttons: [
+                {
+                    text: "Rename",
+                    icon: "ui-icon-edit",
+                    click: () => {
+                        let renameDialog = $('<div class="dialog"><input type="text" id="renameNote"></div>');
+                        renameDialog.dialog({
+                            title: "Renaming: '" + this.name + "'",
+                            autoOpen: true,
+                            width: 'auto',
+                            buttons: [
+                                {
+                                    text: "Confirm",
+                                    icon: "ui-icon-check",
+                                    click: () => {
+                                        let editName = renameDialog.children().eq(0);
+
+                                        this.name = <string> editName.val();
+                                        this.dialog.dialog({
+                                            title: this.name
+                                        });
+
+                                        $('#menu-notes button[note-id=' + this.uuid + ']').html(this.name);
+                                        this.updateNote();
+
+                                        renameDialog.dialog("close");
+                                    }
+                                },
+                                {
+                                    text: "Cancel",
+                                    icon: "ui-icon-cross",
+                                    click: function() {
+                                        $(this).dialog("close");
+                                    }
+                                },
+                            ]
+                        });
+                    },
+                    showText: false
+                },
+                {
+                    text: "Delete",
+                    icon: "ui-icon-trash",
+                    click: () => {
+                        $('#menu-notes button[note-id="' + this.uuid + '"]').remove();
+                        gameManager.notes.remove(this.uuid);
+                        this.dialog.dialog("close");
+                        this.dialog.remove();
+                        socket.emit("deleteNote", this.uuid);
+                    },
+                    showText: false
+                }
+            ]
+        });
+        this.dialog.children().eq(0).bind('input propertychange', () => {
+            this.text = <string> this.dialog.children().eq(0).val();
+            this.updateNote();
+        });
+    }
+
+    updateNote() {
+        socket.emit("updateNote", this.as_dict());
+    }
+
+    show(): void {
+        this.dialog.dialog({title: this.name});
+        this.dialog.children().eq(0).val(this.text);
+        this.dialog.dialog("open");
+    }
+
+    as_dict(): Dictionary<string> {
```

Comment:
I would suggest changing this to `asDict` as typescript uses camel casing and not snake casing like python.